src/prefect/backend/flow_run.py

```diff
@@ -730,7 +730,7 @@ def get_all_task_runs(self) -> List["TaskRunView"]:
         for task_run in task_runs:
             self._cache_task_run_if_finished(task_run)
 
-        return task_runs + list(self._cached_task_runs.values())
+        return list(self._cached_task_runs.values())
```

Comment:
This won't suffice because some of the task runs may not be finished

Reply:
See now. Also, (apologies if this suggestion should go somewhere else, let me know), I think what's still missing here is that child flows are treated differently then the parent flow when you want to get the results of tasks. For example if I have

```
with Flow("parent-flow") as flow2:
    res = create_flow_run(flow_name="mysubflow", project_name="myproject")
    res2 = mytask()

res3 = flow2.run() #prefect.engine.State ()

#to get result of task in parent flow I can use the name of the task in the flow
res3.result[res2].result

# but to get the result of a task in the child flow
subflow_id = res3.result[res].result
frv=FlowRunView.from_flow_run_id(subflow_id)
frv.get_task_run(task_slug="x").get_result()
```

So in one case to get at the result I need to know the assigned variable names in the flow declaration but in the other case I need to know the slugs or task_id's in the registered subflow.  Am I missing something?

Reply:
Yeah unfortunately this is because of how/when task slugs are generated. I originally implemented something that would do the lookup by task object but it doesn't work consistently. This is something that we want to polish in the future but there are some lower-level change necessary first.